app/assets/main/react_components/User/ClimateActionsContainer.jsx

```diff
@@ -1,142 +1,165 @@
 
   const formatedUserActions = (inVal) => {
     return inVal.map((userActions) => ({
       ...userActions,
       id: userActions.id.toString(),
-    }))
-  }
+    }));
+  };
 
   const acceptedUserActions = (inVal) => {
     return formatedUserActions(inVal)
       .filter((action) => action.status !== true)
-      .map((action) => ({ ...action }))
-  }
+      .map((action) => ({ ...action }));
+  };
   const doneUserActions = (inVal) => {
     return formatedUserActions(inVal)
       .filter((action) => action.status !== false)
-      .map((action) => ({ ...action }))
-  }
+      .map((action) => ({ ...action }));
+  };
 
   const columnUserActions = (acceptedList, doneActions) => {
     return {
       [1]: {
-        id: 'Accepted',
-        name: 'Your accepted actions:',
+        id: "Accepted",
+        name: "Your accepted actions:",
         items: acceptedList,
       },
       [2]: {
-        id: 'Performed',
-        name: 'Your performed actions:',
+        id: "Performed",
+        name: "Your performed actions:",
         items: doneActions,
       },
-    }
-  }
+    };
+  };
   const [columns, setColumns] = useState(
     columnUserActions(
       acceptedUserActions(totUserActions),
-      doneUserActions(totUserActions),
-    ),
-  )
+      doneUserActions(totUserActions)
+    )
+  );
   //******************************************************* */
 
   const localActionsWithUserActions = JSON.parse(actionsWithUserActions).map(
     (action) => ({
       ...action,
       accepted: true,
-    }),
-  )
+    })
+  );
 
   const localActionsWithoutUserActions = JSON.parse(
-    actionsWithoutUserActions,
+    actionsWithoutUserActions
   ).map((action) => ({
     ...action,
     accepted: false,
-  }))
+  }));
 
   const totClimateActions = [
     ...localActionsWithoutUserActions,
     ...localActionsWithUserActions,
-  ]
+  ];
 
   const [climateActionsUser, setClimateActionsUser] = useState([
     ...totClimateActions,
-  ])
+  ]);
 
   const updateLocalAccepted = (actionID) => {
     setClimateActionsUser(
       climateActionsUser.map((action) =>
         action.id === actionID
           ? { ...action, accepted: !action.accepted }
-          : action,
-      ),
-    )
-  }
+          : action
+      )
+    );
+  };
+  const monthlyAction = climateActionsUser.find(
+    (action) => action.action_of_the_month === true
+  );
 
-  useEffect(() => {
-    deletedAction != null && updateLocalAccepted(deletedAction)
-  }, [deletedAction])
+  const setCategory = (cat) => {
+    setCurrCategory(cat);
+    const filteredActions = cat
+      ? totClimateActions.filter(
+          (temp) => temp.climate_action_category_id === cat
+        )
+      : totClimateActions;
 
-  const monthlyAction = climateActionsUser.find(
-    (action) => action.action_of_the_month === true,
-  )
+    const filteredActionsWithStatus = filteredActions.map((x) => {
+      return totUserActions.some((y) => y.climate_action_id === x.id)
```

Comment:
A bit hard to interpret x and y, maybe change name to something including the word "action"